P6 cl 178 tooltips for role tags ffx

f2xiao opened the pull request:
This PR adds tooltips for filter role tag, the changes include:
✅add icons files under src/assets/icons directory
✅add data.ts file under src/utils directory 
✅add a component`TooltipWrapper` under src/components/Tooltip directory

The hi-fi styling will be worked after this hi-fi filter [ticket](https://makeitmvp.atlassian.net/browse/P6CL-167) is merged

Note: Originally in this PR, I added tooltip component to the filter component,  since Kevin is working on that file with his ticket at the same time I have reverted the `Filter` component file back to match the `develop` branch

How to test:
replace all the code in file src/components/Filter/Filter.tsx with the following code:
```
import { Card, CardContent, CardDescription, CardHeader } from '../ui/card'
import { cn } from '../lib/utils'
import RoleBadge from './RoleBadge'
import { roleItems } from '@/utils/data'
import { TooltipWrapper } from '../Tooltip/TooltipWrapper'

interface filterProps {
  filterGoalBuddies: Function
  filter: {
    mentor: boolean
    accountability: boolean
    networking: boolean
  }
}

const Filter: React.FC<filterProps> = ({ filterGoalBuddies, filter }) => {
  return (
    <Card
      className={cn(
        'w-64 h-44 pb-0 border-black rounded-md border-b-2 border-r-2',
      )}
    >
      <CardHeader className={cn('pt-2 pr-4')}>
        <CardDescription
          className={cn('text-right font-light m-0 cursor-pointer')}
          onClick={() => filterGoalBuddies('')}
        >
          Clear All
        </CardDescription>
      </CardHeader>
      <CardContent className={cn('flex flex-col gap-4 pr-4 pl-4 pb-0')}>
        <div className="flex flex-row justify-between relative">
          <TooltipWrapper roleItem={roleItems[0]}>
            <label>
              <RoleBadge colour={'bg-blue-600'} />
              Mentor
            </label>
          </TooltipWrapper>
          <span
            className={
              filter.mentor
                ? cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black',
                  )
                : cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white',
                  )
            }
            onClick={() => {
              filterGoalBuddies('mentor')
            }}
          />
        </div>
        <div className="flex flex-row justify-between relative">
          <TooltipWrapper roleItem={roleItems[1]}>
            <label>
              <RoleBadge colour={'bg-amber-500'} />
              Goal Buddy
            </label>
          </TooltipWrapper>
          <span
            className={
              filter.accountability
                ? cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black',
                  )
                : cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white',
                  )
            }
            onClick={() => {
              filterGoalBuddies('accountability')
            }}
          />
        </div>
        <div className="flex flex-row justify-between">
          <TooltipWrapper roleItem={roleItems[2]}>
            <label>
              <RoleBadge colour={'bg-green-600'} />
              Networking
            </label>
          </TooltipWrapper>
          <span
            className={
              filter.networking
                ? cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black',
                  )
                : cn(
                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white',
                  )
            }
            onClick={() => {
              filterGoalBuddies('networking')
            }}
          />
        </div>
      </CardContent>
    </Card>
  )
}

export default Filter
```
hover over the role tag, the tooltip should show up as the following image:
![image](https://github.com/user-attachments/assets/339eb659-ba4f-45c0-bf88-af07472a053f)
![image](https://github.com/user-attachments/assets/ce3ed3db-47ef-47ae-a156-a9073f5515de)
![image](https://github.com/user-attachments/assets/65cdd836-83f9-4443-beed-1d28e696aa4e)

AD9-1 commented:
> Make sure you have a done a pull and merge from develop. You are not getting any database info and so no users are showing and this has been resolved.
> 
> The ticket description looks like this tool tip hover should also be added to the profile labels in the edit user profile. Am I misunderstanding or is there another ticket to add it to that location?
> 
> Tooltip is working in the filter :)
> 
>  filter_tooltip_test.mp4

Tootip will only show in the filter component, not userprofilemodal or edit profile modal

AD9-1 commented:
The background overlay appears when hovering over filter options, but it does not affect them.

robester0403 commented on src/components/Filter/Filter.tsx:
```diff
@@ -1,64 +1,146 @@
+        </div>
+        <div className="flex flex-row justify-between relative">
+          <TooltipWrapper
+            content={
```
for all 3 icons, line 71 to 83 should be it's own component instead of repeating it 3 times.

you can have a component called `FilterTooltipContent`

it should look like this and this would be the component that's not reused in all of the ui

```
...
content = {<FilterTooltipContent ... (whatever props you need) />}
...
```

robester0403 commented on src/components/Tooltip/TooltipContentWrapper.tsx:
```diff
@@ -0,0 +1,17 @@
+  className?: string // Allow styling if needed
+}
+
+export function TooltipContentWrapper({
```
this wrapper is only for styling? Can you not just put a div around content in the tooltip wrapper?

f2xiao commented on src/components/Filter/Filter.tsx:
```diff
@@ -1,64 +1,146 @@
+import { TooltipContentWrapper } from '../Tooltip/TooltipContentWrapper'
+import { roleItems } from '../../utils/data'
 
 interface filterProps {
```
filter related code is @KevinMeisenbacher

f2xiao commented on src/components/Filter/Filter.tsx:
```diff
@@ -1,64 +1,146 @@
-        networking: boolean
-    }
+  filterGoalBuddies: Function
+  filter: {
```
agree, @KevinMeisenbacher

f2xiao commented on src/components/Filter/Filter.tsx:
```diff
@@ -1,64 +1,146 @@
+        </CardDescription>
+      </CardHeader>
+      <CardContent className={cn('flex flex-col gap-4 pr-4 pl-4 pb-0')}>
+        <div className="flex flex-row justify-between relative">
```
yeah, DRY is the way @KevinMeisenbacher

JeffreyChuCPA replied:
Appears to have been changed in his PR now

KevinMeisenbacher commented on src/components/Tooltip/TooltipContentWrapper.tsx:
```diff
@@ -0,0 +1,17 @@
+interface TooltipContentWrapperProps {
```
The tooltip is actually single-use - a child of the filter component. I don't see much sense, either, in building it as a separate component if it's not going to be used anywhere else.

AD9-1 replied:
I agree with @KevinMeisenbacher

f2xiao commented:
Note: Originally in this PR, I added tooltip component to the filter component file, however Kevin is working on that file with his ticket at the same time so I have reverted the Filter component file back to match the develop branch to aviod merge conflicts
